doc/Reference/Reactive/state.md

```diff
@@ -164,3 +226,169 @@ This gives you the ability to update a _state_, including the metadata.
+    }
+```
+
+1. Replace all child elements in the _MainPage.xaml_ with the following:
+
+```xml
+    <Page
+        x:Class="SliderApp.MainPage"
+        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+        xmlns:local="using:SliderApp">
+        <Page.DataContext>
+            <local:SliderViewModel />
+        </Page.DataContext>
+
+        <StackPanel>
+            <StackPanel Orientation="Horizontal" Spacing="5">
+                <TextBlock Text="Current state value:" />
+                <TextBlock Text="{Binding SliderValue}" />
+            </StackPanel>
+
+            <Border Height="1" Background="DarkGray" />
+
+            <TextBlock Text="Set state value:"/>
+            <Slider Value="{Binding SliderValue, Mode=TwoWay}" />
+        </StackPanel>
+    </Page>
+```
+
+In this scenario, the `DataContext` is set to an instance of the `SliderViewModel` class, which is the generated ViewModel for the `SliderModel` record.
+
+1. When you run the app, moving the `Slider` instantly affects the upper `TextBox`. The `Silder.Value` property has a two-way binding with the `SliderValue` State, so any change to the Slider immediately updates the State value, which in turn affects the data-bound `TextBlock` on top:
+
+    ![A video of the previous slider app in action](/Reference/Reactive/Assets/SliderApp-1.gif)
+
+### Change data of a state
+
+#### Update
+
+To manually update the current value of a state, use its `UpdateAsync` method.
+
+In this example we'll add the method `IncrementSlider` that gets the current value and increases it by one (if it doesn't exceed 100):
+
+```csharp
+    public async ValueTask IncrementSlider(CancellationToken ct = default)
+    {
+        static double incrementValue(double currentValue) =>
+            currentValue <= 99
+            ? currentValue + 1
+            : 1;
+    
+        await SliderValue.UpdateAsync(updater: incrementValue, ct);
+    }
+```
+
+The `updater` parameter of the `Update` method accepts a `Func<T, T>`. The input parameter provides the current value of the State when called. The return value is the new value that will be applied as the new value of the State, in our case we use the `incrementValue` [local function](https://learn.microsoft.com/dotnet/csharp/programming-guide/classes-and-structs/local-functions) to increment `currentValue` by one (or return `1` if the value exceeds `100`).
+
+#### Set
+
+There are additional methods that update the data of a State such as `Set` and `UpdateMessage`, explained [here](xref:Uno.Extensions.Reactive.State#update-how-to-update-a-state). The `Set` method is the same as the `Update`, except that in `Set` there is no callback that provides the current value, instead a new value is provided directly and the old value is discarded:
+
+```csharp
+public async ValueTask SetSliderMiddle(CancellationToken ct = default)
+{
+    await SliderValue.SetAsync(50, ct);
+}
+```
```

Comment:
There is already a section above this one that is for `Set`. We should remove that old one if this one here should be the right one